### PR TITLE
add 117 Model server

### PR DESCRIPTION
**Pull request by OpenVINO-dev-contest**
A simple example for model server deployment.
https://github.com/openvinotoolkit/openvino_notebooks/issues/784

### CODE REVIEW

**Comment by ryanloney** [comment]
@dtrawins @mzegla can you please review the notebook? If anyone else from OVMS team has feedback or suggestions, please share it. Thank you!

File: notebooks/117-model-server/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Introduction to OpenVINO™ Model Server
+
+This notebook demonstrates how to shows how to deploy a model server and request prediction from it.
```
Comment (dtrawins) [inline notebooks/117-model-server/README.md:3]:
```suggestion
This notebook demonstrates how to deploy a model server and request predictions from a client application.
```

File: notebooks/117-model-server/117-model-server.ipynb [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+    "!docker run hello-world"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "85b48949",
+   "metadata": {},
+   "source": [
+    "## Step 2: Download the Model Server\n",
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:67]: If we would like to reduce the number of steps, this one could be dropped. OVMS image will be pulled with `docker run` command in step 4, if it's not on the machine already.

File: notebooks/117-model-server/117-model-server.ipynb [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4fc2e171",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker run -d --rm  --name=\"ovms\" -v $(pwd)/models:/models -p 9000:9000 openvino/model_server:2022.3 \\\n",
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:190]: Did you consider using tag `latest`? This way the version will not have to be manually updated with upcoming releases.

File: notebooks/117-model-server/117-model-server.ipynb [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+   "execution_count": 2,
+   "id": "9230a63a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "path = Path(\"models/detection/1\")\n",
+    "if not path.exists():\n",
+    "    !mkdir -p models/detection/1\n",
+    "!cp ../004-hello-detection/model/* models/detection/1"
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:146]: Do we need the Python code? `mkdir -p` should not return error if path exists, so do we need to explicitly check it?

File: notebooks/117-model-server/117-model-server.ipynb [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+   "execution_count": null,
+   "id": "32681eb6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "path = Path(\"models/detection/1\")\n",
+    "if not path.exists():\n",
+    "    !mkdir -p models/detection/1\n",
+    "!mkdir models\\detection\\1\n",
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:169]: Same as above and there are also two mkdir calls.

File: notebooks/117-model-server/117-model-server.ipynb [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4fc2e171",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker run -d --rm  --name=\"ovms\" -v $(pwd)/models:/models -p 9000:9000 openvino/model_server:2022.3 \\\n",
+    "--model_path /models/detection/ --model_name detection --port 9000 --shape auto"
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:191]: Does the demo use reshaping feature? Is the input data the same as the model input? If yes, then `--shape auto` could be removed for simplicity.

File: notebooks/117-model-server/117-model-server.ipynb [hunk 7]
```diff
@@ -0,0 +1,549 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "942cf5ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker stop ovms"
```
Comment (mzegla) [inline notebooks/117-model-server/117-model-server.ipynb:508]: We could also run `docker rm` after that.

File: notebooks/README.md [hunk 8]
```diff
@@ -93,7 +93,8 @@ More amazing notebooks here!
...
 | [114-quantization-simplified-mode](114-quantization-simplified-mode/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?labpath=notebooks%2F114-quantization-simplified-mode%2F114-quantization-simplified-mode.ipynb) | Quantize Image Classification Models with POT in Simplified Mode|
 | [115-async-api](115-async-api/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?labpath=notebooks%2F115-async-api%2F115-async-api.ipynb) | Use Asynchronous Execution to Improve Data Pipelining| |
-| [116-sparsity-optimization](116-sparsity-optimization/)| Improve performance of sparse Transformer models |
+| [116-sparsity-optimization](116-sparsity-optimization/)<br> | Improve performance of sparse Transformer models |
+| [117-model-server](notebooks/117-model-server/)| A basic introduction to using OpenVINO™ Model Server (OVMS) |
```
Comment (ryanloney) [inline notebooks/README.md:97]:
```suggestion
| [117-model-server](notebooks/117-model-server/)| Introduction to model serving with OpenVINO™ Model Server (OVMS) |
```

File: README.md [hunk 9]
```diff
@@ -93,7 +93,8 @@ More amazing notebooks here!
...
 | [114-quantization-simplified-mode](notebooks/114-quantization-simplified-mode/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?labpath=notebooks%2F114-quantization-simplified-mode%2F114-quantization-simplified-mode.ipynb) | Quantize Image Classification Models with POT in Simplified Mode|
 | [115-async-api](notebooks/115-async-api/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?labpath=notebooks%2F115-async-api%2F115-async-api.ipynb) | Use Asynchronous Execution to Improve Data Pipelining| |
-| [116-sparsity-optimization](notebooks/116-sparsity-optimization/)| Improve performance of sparse Transformer models |
+| [116-sparsity-optimization](notebooks/116-sparsity-optimization/)<br> | Improve performance of sparse Transformer models |
+| [117-model-server](notebooks/117-model-server/)| A basic introduction to using OpenVINO™ Model Server (OVMS) |
```
Comment (ryanloney) [inline README.md:97]:
```suggestion
| [117-model-server](notebooks/117-model-server/)| Introduction to model serving with OpenVINO™ Model Server (OVMS) |
```

File: notebooks/117-model-server/117-model-server.ipynb [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "01412caf",
+   "metadata": {},
+   "source": [
+    "# Hello Model Server\n",
+    "\n",
+    "A basic introduction to using OpenVINO™ Model Server (OVMS).\n",
```
Comment (ryanloney) [inline notebooks/117-model-server/117-model-server.ipynb:11]:
```suggestion
    "Introduction to OpenVINO™ Model Server (OVMS).\n",
```

File: notebooks/117-model-server/117-model-server.ipynb [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+    "\n",
+    "## What is Model Server ?\n",
```
Comment (ryanloney) [inline notebooks/117-model-server/117-model-server.ipynb:13]:
```suggestion
    "## What is Model Serving?\n",
```

File: notebooks/117-model-server/117-model-server.ipynb [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+    "Model Server hosts models and makes them accessible to software components over standard network protocols: a client sends a request to the model server, which performs model inference and sends a response back to the client. Model Server offers many advantages for efficient model deployment:\n",
```
Comment (ryanloney) [inline notebooks/117-model-server/117-model-server.ipynb:14]:
```suggestion
    "A model server hosts models and makes them accessible to software components over standard network protocols. A client sends a request to the model server, which performs inference and sends a response back to the client. Model serving offers many advantages for efficient model deployment:\n",
```

File: notebooks/117-model-server/117-model-server.ipynb [hunk 13]
```diff
@@ -0,0 +1,530 @@
...
+    "```\n",
+    "\n",
+    "\n",
+    "* Each model should be stored in a dedicated directory, e.g. model1 and model2.\n",
+    "\n",
+    "* Each model directory should include a sub-folder for each of its versions (1,2, etc). The versions and their folder names should be positive integer values.\n",
+    "\n",
+    "* Note: In execution, the versions are enabled according to a pre-defined version policy. If the client does not specify the version number in parameters, by default, the latest version is served.\n",
+    "\n",
+    "* Every version folder must include model files, that is, .bin and .xml for IR, .onnx for ONNX, .pdiparams and .pdmodel for Paddlepaddle. The file name can be arbitrary.\n"
```
Comment (ryanloney) [inline notebooks/117-model-server/117-model-server.ipynb:103]:
```suggestion
    "* Every version folder must include model files, that is, `.bin` and `.xml` for OpenVINO IR, `.onnx` for ONNX, `.pdiparams` and `.pdmodel` for Paddle Paddle, and `.pb` for TensorFlow. The file name can be arbitrary.\n"
```

**Review by ryanloney — CHANGES_REQUESTED** [review]
A few minor edits

File: notebooks/117-model-server/117-model-server.ipynb (hunk 3 above)
Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb]: done

File: notebooks/117-model-server/117-model-server.ipynb (hunk 4 above)
Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb:126]: I think Python can be used to ensure whether the Path exits for both Win and Linux os

File: notebooks/117-model-server/117-model-server.ipynb (hunk 2 above)
Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb]: done

File: notebooks/117-model-server/117-model-server.ipynb (hunk 5 above)
Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb]: Fixed

File: notebooks/117-model-server/117-model-server.ipynb (hunk 6 above)
Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb]: done

File: notebooks/117-model-server/117-model-server.ipynb (hunk 4 above)
Comment (mzegla) [reply notebooks/117-model-server/117-model-server.ipynb:126]: In that case, what do you think about using only Python in that cell? If you used packages like `os` or `shutil`, you would have only Python code that can be used on multiple OS, so there will be no need to have separate cells for Linux&Mac and for Windows.

Comment (OpenVINO-dev-contest) [reply notebooks/117-model-server/117-model-server.ipynb]: Thanks for your suggestion. the code is updated

**Review by mzegla — APPROVED** [review]

**Review by ryanloney — APPROVED** [review]

**Pull request merged**